### PR TITLE
revert: use durable objects for sync ratelimiting

### PR DESCRIPTION
**Pull request by chronark**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new Durable Object-based rate limiting mechanism.
	- Added a new property for configuring rate limiting behavior.
	- Implemented a new `DurableRateLimiter` class for managing rate limits.
	- Added a `DurableObjectRatelimiter` class for managing rate limiting using durable object state.
	- Enhanced rate limiting configurations with more granular control.

- **Bug Fixes**
	- Updated rate limiting implementation to improve accuracy and performance.

- **Refactor**
	- Replaced agent-based rate limiter with a Durable Object rate limiter.
	- Restructured rate limiting client and object implementations.

- **Tests**
	- Expanded rate limiting test coverage with multiple new test scenarios.
	- Added tests for various rate limit configurations and edge cases.

### CODE REVIEW

File: apps/api/src/pkg/ratelimit/do_client.ts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+          this.logger.warn("calling the ratelimit DO failed, retrying ...", {
+            identifier: req.identifier,
+            error: (e as Error).message,
+          });
+
+          return this.getStub(req.objectName).fetch(url, {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({ reset: req.reset }),
+          });
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/api/src/pkg/ratelimit/do_client.ts:214]: good catch
